addons/mail/static/tests/new/messaging/messaging_tests.js

```diff
@@ -0,0 +1,83 @@
+/** @odoo-module **/
+
+let target;
+QUnit.module("messaging", {
+    async beforeEach() {
+        target = getFixture();
+    },
+});
+import { afterNextRender, start, startServer } from "@mail/../tests/helpers/test_utils";
+
+import { getFixture } from "@web/../tests/helpers/utils";
+
+QUnit.test(
+    "Posting a message to a partner out of discuss should open a chat window",
+    async function (assert) {
+        const pyEnv = await startServer();
+        const resPartnerId = pyEnv["res.partner"].create({ name: "Dumbledore" });
+        const mailChannelId = pyEnv["mail.channel"].create({
+            channel_member_ids: [
+                [0, 0, { partner_id: pyEnv.currentPartnerId }],
+                [0, 0, { partner_id: resPartnerId }],
+            ],
+            channel_type: "chat",
+        });
+        const messageId = pyEnv["mail.message"].create({
+            author_id: resPartnerId,
+            body: "You're a wizard, harry.",
+            model: "mail.channel",
+            res_id: mailChannelId,
+        });
+        const [message] = pyEnv["mail.message"].searchRead([["id", "=", messageId]]);
+        message.author = { name: message.author_id[1], id: message.author_id[0] };
+        await start();
+        await afterNextRender(() =>
+            pyEnv["bus.bus"]._sendone("channel-1", "mail.channel/new_message", {
+                id: message.id,
+                message,
+            })
+        );
```

Comment:
actually all of this could be `/mail/chat_post` with `mockedUserId` âram

Reply:
updated :+1: